src/app/vm/vm-sidebar/affinity-group-selector/affinity-group-selector.component.ts

```diff
@@ -1,80 +1,119 @@
+    private store: Store<State>,
     private affinityGroupService: AffinityGroupService,
     private dialogRef: MatDialogRef<AffinityGroupSelectorComponent>,
     private dialogService: DialogService,
-    @Inject(MAT_DIALOG_DATA) public data
+    private formBuilder: FormBuilder,
   ) {
-    this.vm = data.vm;
-  }
-
-  public ngOnInit(): void {
-    this.loadGroups();
   }
 
-  public get groupName(): string {
-    return this.vm.affinityGroup.length && this.vm.affinityGroup[0].name;
+  public ngOnChanges(changes: SimpleChanges) {
+    const affinityGroup = changes.affinityGroups.currentValue;
+    if (affinityGroup) {
+      this.affinityGroups = affinityGroup;
+      this.sortList();
+    }
   }
 
-  public get groupNames(): Array<string> {
-    return this.affinityGroups.map(_ => _.name);
-  }
-
-  public get selectedAffinityGroup(): AffinityGroup {
-    return this.affinityGroups.find(_ => _.name === this.selectedAffinityGroupName);
+  public ngOnInit(): void {
+    this.createForm();
+    this.sortList()
   }
 
-  public createGroup(name: string): void {
-    this.affinityGroupService
-      .create({
-        name,
-        type: AffinityGroupType.hostAntiAffinity
-      })
-      .subscribe(
-        affinityGroup => this.dialogRef.close(affinityGroup.id),
-        error => this.dialogService.alert({ message: error.message })
-      );
+  public createGroup(): void {
+    const newAffinityGroup = {
+      name: this.affinityGroupForm.value.name,
+      type: this.affinityGroupForm.value.type,
+      description: this.affinityGroupForm.value.description
+    } as AffinityGroup;
+    this.onCreateAffinityGroup.emit(newAffinityGroup);
   }
 
-  public changeGroup(name: string): void {
-    this.selectedAffinityGroupName = name;
-    this.dialogRef.close(this.selectedAffinityGroup.id);
+  public changeGroup(group: AffinityGroup): void {
+    this.selectedGroup = group;
   }
 
-  public removeGroup(): void {
-    this.dialogRef.close('');
+  public submit(): void {
+    const selectedGroupIds = this.preselectedAffinityGroups.map(group => group.id);
+    selectedGroupIds.push(this.selectedGroup.id);
+    const groupIds = selectedGroupIds.join();
+    this.onSubmit.emit(groupIds);
   }
 
-  public onCancel(): void {
-    this.dialogRef.close();
+  private sortList() {
+    this.list = this.affinityGroups.map(group => {
+      const isPreselected = !!this.preselectedAffinityGroups.find(preselected => preselected.id === group.id);
+      return {...group, isPreselected}
+    });
+    this.list.sort(sortBySelected)
   }
 
-  private loadGroups(): void {
-    this.loading = true;
-    this.affinityGroupService.getList().pipe(
-      finalize(() => this.loading = false))
-      .subscribe(groups => {
-        this.affinityGroups = groups;
-      });
+  private createForm() {
+    this.affinityGroupForm = this.formBuilder.group({
+      name: this.formBuilder.control('', [
+        Validators.required,
+        Validators.maxLength(63),
```

Comment:
We have `maxEntityNameLength = 63` in `vm-creation.component.ts`, consider using a constant.